samples/contrib/kubeflow-e2e-mnist/kubeflow-e2e-mnist-kfpv2.ipynb

```diff
@@ -0,0 +1,777 @@
+    "\n",
+    "**Note**: Run the following command in your terminal, replace `<kubeflow-user-project>` with your Kubeflow project name:\n",
+    "\n",
+    "```\n",
+    "cat <<EOF | kubectl create -f -\n",
+    "apiVersion: v1\n",
+    "kind: PersistentVolumeClaim\n",
+    "metadata:\n",
+    "  name: 'workflow1-model-volume'\n",
+    "  namespace: '<kubeflow-user-project>'\n",
+    "spec:\n",
+    "  accessModes:\n",
+    "  - ReadWriteOnce\n",
+    "  resources:\n",
+    "    requests:\n",
+    "      storage: 1Gi\n",
+    "EOF\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Define the Pipelines tasks\n",
+    "\n",
+    "To run this Pipeline, you should define:\n",
+    "1. Katib hyperparameter tuning\n",
+    "2. TFJob training\n",
+    "3. KServe inference\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Import necessary modules"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp\n",
+    "import kfp.dsl as dsl\n",
+    "from kfp import components\n",
+    "from kfp.dsl import Output, Artifact\n",
+    "\n",
+    "from kubeflow.katib import ApiClient\n",
+    "from kubeflow.katib import V1beta1ExperimentSpec\n",
+    "from kubeflow.katib import V1beta1AlgorithmSpec\n",
+    "from kubeflow.katib import V1beta1ObjectiveSpec\n",
+    "from kubeflow.katib import V1beta1ParameterSpec\n",
+    "from kubeflow.katib import V1beta1FeasibleSpace\n",
+    "from kubeflow.katib import V1beta1TrialTemplate\n",
+    "from kubeflow.katib import V1beta1TrialParameterSpec\n",
+    "\n",
+    "from typing import Dict, List, NamedTuple"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### Step 1. Katib hyperparameter tuning task\n",
+    "\n",
+    "Create the Kubeflow Pipelines task for the Katib hyperparameter tuning. This Experiment uses \"random\" algorithm and TFJob for the Trial's worker.\n",
+    "\n",
+    "The Katib Experiment is similar to this example: https://github.com/kubeflow/katib/blob/master/examples/v1beta1/kubeflow-training-operator/tfjob-mnist-with-summaries.yaml."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# You should define the Experiment name, namespace and number of training steps in the arguments.\n",
+    "@dsl.component(\n",
+    "    base_image='python:3.8',\n",
+    "    packages_to_install=['kubeflow-katib==0.12.0']\n",
+    ")\n",
+    "def create_katib_experiment_task(experiment_name: str, experiment_namespace: str, training_steps: str\n",
+    "                                ) -> NamedTuple('Outputs', [('experiment_spec_json', Dict[str, str])]):\n",
```

Comment:
`NamedTuple` with one output should probably just be a single output (`-> Dict[str, str]`)

This simplifies the component data passing logic when constructing the pipeline from `task.outputs['experiment_spec_json']` to `task.output`.

Reply:
Discussed offline. Prefer the named output for this example.